.github/workflows/continuous-integration.yml

```diff
@@ -12,11 +12,106 @@ on:
+    name: "Coding Standards"
 
     runs-on: ubuntu-latest
 
     steps:
-      - name: "Hello"
-        run: echo "Hello, François and Pim!"
+      - name: "Checkout"
+        uses: actions/checkout@v1.2.0
+
+      - name: "Install PHP with extensions"
+        uses: shivammathur/setup-php@1.5.5
+        with:
+          coverage: none
+          extension-csv: "intl"
+          php-version: 7.2
+
+      - name: "Cache dependencies installed with composer"
+        uses: actions/cache@v1.0.3
+        with:
+          path: ~/.composer/cache
+          key: php7.2-composer-${{ hashFiles('**/composer.json') }}
+          restore-keys: |
+            php7.2-composer-
+
+      - name: "Install dependencies with composer"
+        run: $(which composer) install --no-interaction --no-progress --no-suggest
+
+      - name: "Run squizlabs/php_codesniffer"
+        run: vendor/bin/phpcs --standard=PSR2 src -n
+
+  tests:
+    name: "Tests"
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php-version:
+          - 7.2
+          - 7.3
+          - 7.4
+
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v1.2.0
+
+      - name: "Install PHP with extensions"
+        uses: shivammathur/setup-php@1.5.5
+        with:
+          coverage: none
+          extension-csv: "intl"
+          php-version: ${{ matrix.php-version }}
+
+      - name: "Cache dependencies installed with composer"
+        uses: actions/cache@v1.0.3
+        with:
+          path: ~/.composer/cache
+          key: php${{ matrix.php-version }}-composer-${{ hashFiles('**/composer.json') }}
+          restore-keys: |
+            php${{ matrix.php-version }}-composer-
+
+      - name: "Install dependencies with composer"
+        run: $(which composer) install --no-interaction --no-progress --no-suggest
+
+      - name: "Run tests with phpunit/phpunit"
+        run: vendor/bin/phpunit
+
+  code-coverage:
+    name: "Code Coverage"
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v1.2.0
+
+      - name: "Install PHP"
+        uses: shivammathur/setup-php@1.5.5
+        with:
+          coverage: xdebug
+          php-version: 7.4
+
+      - name: "Cache dependencies installed with composer"
+        uses: actions/cache@v1.0.3
+        with:
+          path: ~/.composer/cache
+          key: php7.4-composer-locked-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            php7.4-composer-locked-
+
+      - name: "Install locked dependencies with composer"
+        run: $(which composer) install --no-interaction --no-progress --no-suggest
+
+      - name: "Create build directory"
+        run: mkdir -p build/logs
+
+      - name: "Collect code coverage with Xdebug and phpunit/phpunit"
+        run: vendor/bin/phpunit --coverage-clover=build/logs/clover.xml
+
+      - name: "Send code coverage report to Codecov.io"
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
This token needs to be added via https://github.com/fzaninotto/faker/settings/secrets.

Also take note of

- https://github.com/codecov/codecov-bash/issues/224
- https://community.codecov.io/t/whitelist-github-action-servers-to-upload-without-a-token/491/8